### PR TITLE
[Refactor] [Step 2/2]Change the name from master to leader

### PR DESCRIPTION
**Pull request by gengjun-git**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Change the master related code to leader.
Split the task into two steps: 
1. change the name from master to leader in the internal code paths.
2. change the name from master to leader in the interface.

The advantage of this is that we only need to rollback the step 2 if If there are compatibility problems.

This is PR is the step 2 of the task. 
1. remove the `isMaster` attribute from the result of `show frontends`.
2. add `forward_to_leader` session variable to replace the `forward_to_master`. But we still remain `forward_to_master` for compatibility.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 11 (18.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|0|1|00.00%|[200]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|0|1|00.00%|[102]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|0|1|00.00%|[80]|
|:large_blue_circle:|com/starrocks/analysis/ShowBrokerStmt.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/analysis/ShowTabletStmt.java|0|1|00.00%|[258]|
|:large_blue_circle:|com/starrocks/analysis/ShowResourcesStmt.java|0|1|00.00%|[41]|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|1|3|33.33%|[105, 107]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[740]|

**Review by padmejin — COMMENTED** [review]
There are some thrift messages defined with "Master". I suggest we mark it deprecated and redefine a new message.

File: fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java [hunk 1]
```diff
@@ -451,6 +452,9 @@ public class SessionVariable implements Serializable, Writable, Cloneable {
     @VariableMgr.VarAttr(name = FORWARD_TO_MASTER)
     private boolean forwardToMaster = false;
 
+    @VariableMgr.VarAttr(name = FORWARD_TO_LEADER)
```
[the remark this answers is not in the document]
Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/qe/SessionVariable.java]: done

**Comment by gengjun-git** [comment]
> There are some thrift messages defined with "Master". I suggest we mark it deprecated and redefine a new message.

Change the thrift message is a very complex task, because we should resolve the compatibility problem for different version of client and sever. We'd better do not change these classes. Eliminating all master words is a `try our best` job.

**Review by padmejin — DISMISSED** [review]

**Comment by nshangyiming** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java [hunk 2]
```diff
@@ -45,7 +46,7 @@ public class FrontendsProcNode implements ProcNodeInterface {
 
     public static final ImmutableList<String> TITLE_NAMES = new ImmutableList.Builder<String>()
             .add("Name").add("IP").add("EditLogPort").add("HttpPort").add("QueryPort").add("RpcPort")
-            .add("Role").add("IsMaster").add("ClusterId").add("Join").add("Alive").add("ReplayedJournalId")
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java:48]: why remove `add("IsMaster")`?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/common/proc/FrontendsProcNode.java:48]: to eliminate master words

**Comment by gengjun-git** [comment]
run starrocks_fe_unittest

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Review by padmejin — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 2 / 11 (18.18%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaStatusStmt.java|0|1|00.00%|[110]|
|:large_blue_circle:|com/starrocks/analysis/AdminShowReplicaDistributionStmt.java|0|1|00.00%|[69]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|0|1|00.00%|[80]|
|:large_blue_circle:|com/starrocks/analysis/ShowBrokerStmt.java|0|1|00.00%|[59]|
|:large_blue_circle:|com/starrocks/analysis/ShowTabletStmt.java|0|1|00.00%|[258]|
|:large_blue_circle:|com/starrocks/analysis/ShowResourcesStmt.java|0|1|00.00%|[41]|
|:large_blue_circle:|com/starrocks/common/proc/FrontendsProcNode.java|1|3|33.33%|[105, 107]|
|:large_blue_circle:|com/starrocks/qe/SessionVariable.java|1|2|50.00%|[740]|

**Review by chaoyli — APPROVED** [review]

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Comment by gengjun-git** [comment]
run starrocks_admit_test

**Pull request merged**